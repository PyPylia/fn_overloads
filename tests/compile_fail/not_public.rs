#![feature(unboxed_closures, fn_traits)]

mod hidden {
    use fn_overloads::fn_overloads;

    fn_overloads! {
        fn not_public {
            (val: u32) -> u32 { val }
        }
    }
}

fn main() {
    hidden::not_public(1);
}