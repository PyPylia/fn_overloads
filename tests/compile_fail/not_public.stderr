error[E0603]: unit struct `not_public` is private
  --> tests/compile_fail/not_public.rs:14:13
   |
14 |     hidden::not_public(1);
   |             ^^^^^^^^^^ private unit struct
   |
note: the unit struct `not_public` is defined here
  --> tests/compile_fail/not_public.rs:6:5
   |
6  |     fn_overloads! {
   |     ^^^^^^^^^^^^^
   = note: this error originates in the macro `fn_overloads` (in Nightly builds, run with -Z macro-backtrace for more info)
